doc/mgmt/SONiC_overlay_ConfigDB.md

```diff
@@ -0,0 +1,234 @@
+
+# 1 Functional Requirement
+
+## 1.1 swss-common return default value from Yang model overlay
+
+- Return default value is optional.
+  - Application can read config without default value, also can read config with default value.
+- Backward compatibility with existed code and applications.
+
+## 1.2 swss-common return static config from static config overlay
+
+- Static buffer config stored in static config tables.
+- Return static config is optional.
+  - Application can decide read config with static config or not.
+- Backward compatibility with existed code and applications.
+  - For backward compatibility when initialize buffer config from graph, config will be write to both config tables and static config tables.
+  - After code migrate to use static buffer config, static config will only write to static config tables.
+  - Static config support delete/revert operation:
+    - In some user scenario, user need delete a static config, and also may add config back later.
+    - For delete operation, data in static config table will not be delete.
+    - Static overlay use STATIC_CONFIG_DELETE table to handle dlete/revert:
+      - When delete a static config item, static overlay will add the item key to STATIC_CONFIG_DELETE table.
+      - When read static config, all item in STATIC_CONFIG_DELETE will not include in result.
+
+# 2 Design
+
+- Layered config DB design diagram:
+
+<img src="./images/swss-common-layer.png"  />
+
+- API design diagram:
+
+<img src="./images/swss-common-default-value.png"  />
+
+## 2.1 Considerations
+
+### How to get default value
+
+|                                                               | Pros                                              | Cons                                                                                                                                              |
+| ------------------------------------------------------------- | ------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Get default value from Yang model in read API.                | Redis config DB keeps no change.                  | 3 MB memory per-process because need load Yang model and reference libyang.<br>50ms to load yang model.<br>8ms to read default value 10000 times. |
+| Write default value to default value DB when write config DB. | Better read performance, Less memory consumption. | Need add new Redis DB for default value.                                                                                                          |
+
+### How to get static config
+
+- Static config will stored in a new redis database 'STATIC_CONFIG_DB'
+- Static config DB tables will have exactly same name and schema with config DB tables.
+- Data will read form static config DB with swsscommon API.
+- Static config DB will save and persist for warm-reboot and fast-reboot.
```

Comment:
>warm-reboot and fast-reboot [](http://example.com/codeflow?start=45&length=27)

We need to consider more scenarios like cold-reboot, and old schema->new schema upgrade with db migration. In general, each HLD should have a section to cover warm-reboot/fast-reboot.

Reply:
Fixed, added a reboot section.